src/app_service/service/accounts/service.nim

```diff
@@ -344,6 +344,15 @@ QtObject:
       # Waku V1 config
       result["ListenAddr"] = newJString("0.0.0.0:" & wV1Port)
 
+    # Don't override log level on login. For onboarding it is required, nothing to override
+    if login:
```

Comment:
@stefandunca just a minor thing, suggestion:
keep the function as it is (without adding `login` param):
`proc getDefaultNodeConfig*(self: Service, installationId: string, recoverAccount: bool): JsonNode =`

and then in the body we can actually apply all settings from the db if they exists, I agree with you that default config should be used only for creating new accounts and in all other cases values from the default settings should be overwritten with those from the db and as the last step they should be overwritten with values from ENV and as so sent to status go.

```
proc getDefaultNodeConfig*(self: Service, installationId: string, recoverAccount: bool): JsonNode =
  .
  .
  .
  currentLogLevel = self.nodeConfigurationService.getLogLevel()
  if currentLogLevel.len > 0:
    result["LogLevel"] = currentLogLevel

  self.applyEnvValuesToNodeConfig(result)

proc applyEnvValuesToNodeConfig*(self: Service, nodeConfig: var JsonNode) =
  # Source the connection port from the environment for debugging or if default port not accessible
  if existsEnv("STATUS_PORT"):
    let wV1Port = $getEnv("STATUS_PORT")
    # Waku V1 config
    result["ListenAddr"] = newJString("0.0.0.0:" & wV1Port)

  if existsEnv("LOG_LEVEL"):
    let logLvl = getEnv("LOG_LEVEL")
    if logLvl in @["ERROR", "WARN", "INFO", "DEBUG", "TRACE"]:
      result["LogLevel"] = newJString($logLvl)
```

Reply:
> ...
> and then in the body we can actually apply all settings from the db if they exists ...
> ...

I'm not sure I understand your recommendation. It looks to me that here, in `getDefaultNodeConfig`, we don't have access to the DB given that the login was not yet done.
As it looks in status-go we should provide all the node configurations the first time (onboarding) and only updates at login.

Reply:
> we don't have access to the DB given that the login was not yet done

Ah right, the db is not available at that point.

Anyhow I wonder why we're sending the node configuration at all on login at all, actually we should send fields that are set in ENV only if any exist.

Reply:
I don't remember exactly, but I think it was to some issue I saw during the merging of the node config values, that it ended up overwritting some attributes with empty values, so the 'solution' I found was to just send the full node config again. 

I like the idea of sending only the values that change!

Reply:
Good idea, will consider it if ever encounter other issues.